easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -607,11 +608,56 @@ def dependencies(self):
         # each processed easyconfig to remove the unwanted dependencies
         self.log.debug("Dependencies BEFORE filtering: %s" % deps)
         filter_deps = build_option('filter_deps')
+
         if filter_deps:
+            # find if any of the filter_deps are in the form name=version or name=[low:high[
+            complex_fdeps_list = [x for x in filter_deps if "=" in x]
+            complex_fdeps = { }
+            # convert list of k=v into dictionary
+            if complex_fdeps_list:
+                for s in complex_fdeps_list:
```

Comment:
this can be simplified to:

```python
for dep_spec in [x for x in filter_deps if "=" in x]:
```

No need to make it conditional, for loop over empty list won't do anything, and if you only use the `complex_fdeps_list` value in one place using a variable is overkill imho.